components/dashboard/src/teams/TeamUsage.tsx

```diff
@@ -71,9 +77,14 @@ function TeamUsage() {
         })();
     }, [team, startDateOfBillMonth, endDateOfBillMonth, isStartedTimeDescending]);
 
-    if (!BillingMode.showUsageBasedBilling(teamBillingMode)) {
-        return <Redirect to="/" />;
-    }
+    useEffect(() => {
+        if (!teamBillingMode) {
+            return;
+        }
+        if (!BillingMode.showUsageBasedBilling(teamBillingMode)) {
+            window.location.href = gitpodHostUrl.asDashboard().toString();
+        }
+    }, [teamBillingMode]);
```

Comment:
Why does this have to happen in a `useEffect` rather than how it was done before?

Reply:
`teamBillingMode` state is `undefined` on the initial call (which is the first render) of `TeamUsage`, and is only loaded after the first render. Effectively, this means an immediate redirect because the condition was always true.

Prior to this, the redirect was based on a context state:
https://github.com/gitpod-io/gitpod/pull/11814/files#diff-133f74f34700fb70678ded1312ecb51ced6f277b9fba8ddb4b51fb95a45b202dL70